src/resources/js/App.tsx

```diff
@@ -1,18 +1,23 @@
+import React from 'react';
 import { createRoot } from 'react-dom/client';
 import { theme } from './theme';
 import { ThemeProvider } from '@mui/material/styles';
 import { CssBaseline } from '@mui/material';
 import { BrowserRouter } from 'react-router-dom';
 import { Router } from './routes/Router';
+import { AppContext } from './context/AppContext';
 
 export default function App() {
+  const [userName, setUserName] = React.useState('');
   return (
-    <ThemeProvider theme={theme}>
-      <BrowserRouter basename='/main_server/'>
-        <CssBaseline />
-        <Router />
-      </BrowserRouter>
-    </ThemeProvider>
+    <AppContext.Provider value={{ userName, setUserName }}>
+      <ThemeProvider theme={theme}>
+        <BrowserRouter basename='/main_server/'>
```

Comment:
is this just temporary, not all servers start with `/main_server/`

Reply:
You can parse the actual basename from `new URL(baseUrl)`. The baseUrl is global, defined in the base laravel template.

Reply:
we will have to figure out something for that.

Reply:
but otherwise it will default to `/` so we will need to find what the correct permalink is.